web/src/main/java/org/openmrs/module/web/WebModuleUtil.java

```diff
@@ -674,11 +674,12 @@ public static void unloadFilters(Module module) {
 			log.debug("Module: " + module.getModuleId() + " successfully unloaded " + filters.size() + " filters.");
 			moduleFilters.remove(module);
 			
-			for (Iterator<String> i = moduleFiltersByName.keySet().iterator(); i.hasNext();) {
-				String filterName = i.next();
-				Filter filterVal = moduleFiltersByName.get(filterName);
+			
+			for (Entry<String, Filter> entry: moduleFiltersByName.entrySet()) {
+				
+				Filter filterVal = entry.getValue();
 				if (filters.contains(filterVal)) {
-					i.remove();
+					 moduleFiltersByName.remove(entry.getKey());
```

Comment:
It is wrong. You cannot remove elements from a collection that you iterate through using .remove as it will result in a concurrent modification exception. It must be done using the iterator. Please revert.